docs/01-Headless/03-Directives.md

```diff
@@ -1,5 +1,156 @@
+
+    ```
+    focus element clicked <button>​button 1​</button>​
+    ```
+
+    Same if you click on `button 2`, since it also belongs to the element `focus-element` on which we applied the directive.
+
+    ```
+    focus element clicked <button>​button 2</button>​
+    ```
+
+Changing the input does not change anything, because the `update` function has no bindings and it is never called. To solve it, we need the next step.
+
+2.  Observe context changes and call the `update` function accordingly. For example, we want to change the text to the value of the `clickText` input when it changes.
+
+    ```typescript
+    const input = document.getElementById('clickText');
+    input.addEventListener('change', (event) => {
+    	focusElementDirective?.update(event.target.value);
+    });
+    ```
+
+    Now we bound the directive `update` function to a change event on the input `clickText`.
+    Type `Update has been called!` in the input and then click again on `button 1`, you get:
+
+    ```
+    Update has been called! <button>​button 1</button>​
+    ```
+
+    The directive has been correctly updated according to the context change.
+    The last life-cycle event to implement will be the destroy.
+
+3.  Clean up when the `focus-element` gets remove from the DOM. This can be done for example using a `MutationObserver` on the parent DOM element.
+
+    ```typescript
+    const cleanup = (mutations: MutationRecord[]) => {
+    	mutations.forEach((mutation) => {
+    		for (let entry of mutation.removedNodes.entries()) {
+    			if (entry.includes(trackElement)) {
+    				focusElementDirective?.destroy();
+    			}
+    		}
+    	});
+    };
+    const observer = new MutationObserver(cleanup);
+    observer.observe(document.getElementById('content'), {childList: true});
+    ```
+
+    Click on `remove dom element`, that remove completely the `focus-element` container from the DOM. Last log will be printed:
+
+    ```
+    Destroy function executed  focus-element
+    ```
+
+### Frameworks usage
+
+As seen in the previous section, a `Directive` is framework agnostic. But the way frameworks expose the DOM element and binds life-cycle events can be very different. Frameworks controls life-cycle events of the DOM elements, in particular initialization, updates and destroy.
+
+The headless libraries of AgnosUI contain adapters to bind directives in the correct way, so that the corresponding events are called at the right moment benefiting from framework features.
+
+<!-- <framework-specific src="Directives.md"> -->
+
+[Directives in Angular](../../angular/docs/Directives.md),
+[Directives in React](../../react/docs/Directives.md),
+[Directives in Svelte](../../svelte/docs/Directives.md)
+
+<!-- </framework-specific> -->
+
+## Advanced
+
+### Merge directives
+
+If you are using svelte, you can have multiple directives by using multiple time the `use` directive. On other frameworks, this is not the case.
```

Comment:
Very minor: all frameworks should be titlecased, so you could rename here to Svelte (there is also a react one later).